sql/src/main/java/io/crate/expression/operator/CIDROperator.java

```diff
@@ -0,0 +1,115 @@
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.operator;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.network.InetAddresses;
+
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+
+public final class CIDROperator {
+
+    public static final String LLT = Operator.PREFIX + "<<";
+
+    private static final int IPV4_ADDRESS_LEN = 4;
+    private static final int IPV6_ADDRESS_LEN = 16;
+
+    public static void register(OperatorModule module) {
+        module.registerDynamicOperatorFunction(LLT, CmpOperatorResolver.forIpCmpOperator(LLT, CIDROperator::llt));
+    }
+
+    public static boolean llt(String ipStr, String cidr) {
+        if (null == ipStr || null == cidr) {
+            throw new IllegalArgumentException();
+        }
+        boolean leftHasSlash = ipStr.contains("/");
+        boolean rightHasSlash = cidr.contains("/");
+        if (leftHasSlash == rightHasSlash) {
+            return false;
+        }
+        if (false == rightHasSlash || leftHasSlash) {
+            throw new IllegalArgumentException("operands are in the wrong order");
+        }
+        try {
+            BigInteger ip = toBigIntegerIp(InetAddress.getByName(ipStr));
+            return extractRange(cidr, (start, end) -> start.compareTo(ip) <= 0 && ip.compareTo(end) < 0);
+        } catch (UnknownHostException uhe) {
+            throw new IllegalArgumentException(uhe);
+        }
+    }
+
+    public static Tuple<InetAddress, InetAddress> extractRange(String cidr) {
+        return extractRange(cidr, (start, end) -> new Tuple<>(fromBigIntegerIp(start), fromBigIntegerIp(end)));
+    }
+
+    @FunctionalInterface
+    private interface IpRange<R> {
+        R apply(BigInteger cidrs, BigInteger cidre);
+    }
+
+    private static <R> R extractRange(String cidr, IpRange<R> range) {
+        if (null == cidr || false == cidr.contains("/")) {
+            throw new IllegalArgumentException("value must conform with CIDR notation");
+        }
+        Tuple<InetAddress, Integer> tup = InetAddresses.parseCidr(cidr);
+        InetAddress inetAddress = tup.v1();
+        BigInteger base = toBigIntegerIp(inetAddress);
+        BigInteger mask = createMask(inetAddress.getAddress().length, tup.v2());
+        BigInteger start = base.and(mask);
+        BigInteger end = start.add(mask.not());
+        return range.apply(start, end);
+    }
+
+    private static BigInteger toBigIntegerIp(InetAddress address) {
+        return new BigInteger(1, address.getAddress());
+    }
+
+    private static BigInteger toBigIntegerIp(byte[] address) {
+        return new BigInteger(1, address);
+    }
```

Comment:
Maybe inline those? Shallow methods like this often cause more complexity due to the indirection instead of reducing the complexity.